`nupic.core` master build failure on linux due to missing `OS/Path.hpp`

@oxtopus #160 caused `nupic.core` master build failure on linux, leaving only successful mac build. 

The error is:

```
/home/travis/build/numenta/nupic.core/src/examples/regions/helloregions.cpp:28:27: fatal error: nta/OS/Path.hpp: No such file or directory
```

The `OS` here should read `os`, in lower case.

--- a/src/examples/regions/helloregions.cpp
+++ b/src/examples/regions/helloregions.cpp
@@ -25,7 +25,7 @@
 #include <nta/engine/Region.hpp>
 #include <nta/ntypes/Dimensions.hpp>
 #include <nta/ntypes/ArrayRef.hpp>
-#include <nta/OS/Path.hpp>
+#include <nta/os/Path.hpp>
 
 
 

